markers-custom/src/Map.js

```diff
@@ -0,0 +1,85 @@
+import mapboxgl from 'mapbox-gl';
+import React, { useEffect, useRef, useState } from 'react';
+import ReactDOM from 'react-dom';
+import geoJson from './chicago-parks.json';
+import './Map.css';
+
+mapboxgl.accessToken =
+  'pk.eyJ1IjoibWFwYm94IiwiYSI6ImNpejY4M29iazA2Z2gycXA4N2pmbDZmangifQ.-g_vE53SD2WrJ6tFX7QHmA';
+
+const Marker = ({ onClick, children, feature }) => {
+  const _onClick = (e) => {
+    onClick(feature.properties.description);
+  };
+
+  return (
+    <button onClick={_onClick} className="marker">
+      {children}
+    </button>
+  );
+};
+
+const Map = () => {
+  const mapContainerRef = useRef(null);
+
+  const [lng, setLng] = useState(-87.65);
+  const [lat, setLat] = useState(41.84);
+  const [zoom, setZoom] = useState(10);
+
+  // Initialize map when component mounts
+  useEffect(() => {
+    const map = new mapboxgl.Map({
+      container: mapContainerRef.current,
+      style: 'mapbox://styles/mapbox/streets-v11',
+      center: [lng, lat],
+      zoom: zoom,
+    });
+
+    // Render custom marker components
+    geoJson.features.forEach((feature) => {
+      // Create a React ref
+      const ref = React.createRef();
+      // Create a new DOM node and save it to the React ref
+      ref.current = document.createElement('div');
+      // Render a Marker Component on our new DOM node
+      ReactDOM.render(
+        <Marker onClick={markerClicked} feature={feature} />,
+        ref.current
+      );
+
+      // Create a Mapbox Marker at our new DOM node
+      new mapboxgl.Marker(ref.current)
+        .setLngLat(feature.geometry.coordinates)
+        .addTo(map);
+    });
+
+    // Add navigation control (the +/- zoom buttons)
+    map.addControl(new mapboxgl.NavigationControl(), 'top-right');
+
+    map.on('move', () => {
+      setLng(map.getCenter().lng.toFixed(4));
+      setLat(map.getCenter().lat.toFixed(4));
+      setZoom(map.getZoom().toFixed(2));
+    });
+
+    // Clean up on unmount
+    return () => map.remove();
+  }, []); // eslint-disable-line react-hooks/exhaustive-deps
+
+  const markerClicked = (title) => {
+    window.alert(title);
+  };
+
+  return (
+    <div>
+      <div className="sidebarStyle">
+        <div>
+          Longitude: {lng} | Latitude: {lat} | Zoom: {zoom}
+        </div>
```

Comment:
For simplicity, can you drop this feature and logic above from your marker examples as it's not related?